core/src/components/datetime/test/data.spec.ts

```diff
@@ -364,15 +365,15 @@ describe('generateTime()', () => {
         year: undefined,
         hour: 19,
         minute: 50,
-      };
+      } as unknown as DatetimeParts;
```

Comment:
Why do we do `as unknown` here instead of something like 

```suggestion
      } as Partial<DatetimeParts>;
```

Reply:
All downstream consumers of this argument require the entire `DatetimeParts` object. But for what we are testing here we are only using a minimal representation of that object. 

`as unknown as DatetimeParts` allows us to say "we know what we are doing" by forcing the object to a shape that isn't valid to the typings. 

`as Partial<DatetimeParts>` would require a huge overall of the type implementations of all the utilities and assertions that properties are defined before using them. 

I question why we are checking so many things with invalid arguments. That's what typescript gives us for free. But I only resolved the typing errors as part of this.

Reply:
We check invalid arguments because not everyone uses TypeScript. For example, using Vue w/o TypeScript is common enough that we have a page on our docs that show how to remove TypeScript: https://ionicframework.com/docs/vue/quickstart#build-your-way-with-typescript-or-javascript

Reply:
Is this still fairly common in Vue? Our starters and playgrounds both use Typescript. Having unit test coverage on behavior that is driven by the consuming package vs. the implementation of the code is strange to me, especially in utility functions that developers aren't accessing directly. 

If `ion-datetime` can accept potentially a `null` or `undefined` property - I think that makes perfect sense to test & would be possible without manual casting since the type signature supports it.

Reply:
yeah that's fair -- I could see us testing the public APIs with invalid inputs but expect that the internal utils always use valid inputs